addons/binding/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/action/VolvoOnCallActionTypeProvider.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.volvooncall.internal.action;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.smarthome.core.common.registry.ProviderChangeListener;
+import org.openhab.core.automation.Visibility;
+import org.openhab.core.automation.type.ActionType;
+import org.openhab.core.automation.type.ModuleType;
+import org.openhab.core.automation.type.ModuleTypeProvider;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@VolvoOnCallActionTypeProvider } is responsible to give access to
+ * appropriate ModulteTypes
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(immediate = true)
+public class VolvoOnCallActionTypeProvider implements ModuleTypeProvider {
```

Comment:
If you only provide actions, have a look at the "org.openhab.binding.mqtt.action".
You can (and should) define actions that depend on a Thing via annotations.
That leads to shorter code and no fiddling with the ThingRegistry. (I did it like you initially, and the framework guys introduced the ActionHandler as a consequence)